### PR TITLE
Allow emoji to be optional

### PR DESCRIPTION
**Pull request by steven-sheehy**
Emoji icon in release sections would be nice if it was optional in case people wanted a cleaner look and feel for their release notes (like myself).

I've ran `./mvnw verify` and previously signed the CLA

### CODE REVIEW

File: src/main/java/io/spring/releasenotes/generator/ReleaseNotesSection.java [hunk 1]
```diff
@@ -44,7 +45,6 @@
 
 	ReleaseNotesSection(String title, String emoji, List<String> labels) {
 		Assert.hasText(title, "Title must not be empty");
-		Assert.hasText(emoji, "Emoji must not be empty");
```
Comment (snicoll) [inline src/main/java/io/spring/releasenotes/generator/ReleaseNotesSection.java]: This feels a bit odd to have a nullable field in between two mandatory fields. If we were to merge this, the `emoji` parameter should probably move.

**Comment by steven-sheehy** [comment]
Just so I understand, you're advocating removal of the emoji config option and consolidating it into a single title that the user provides, correct? The default section title output would stay the same.

I can do this, I just didn't want to break Spring's tooling or a third party (like [release-notes-generator-action](https://github.com/marketplace/actions/release-notes-generator-action)) that depends upon the current configuration format and has customized the emoji or title.

>> When doing so, we should document that the emoji of a Section is optional rather than silently ignoring it later as you suggest here.

With the approach you recommend, it's more appropriate to say that the title accepts, but doesn't require, an emoji in markdown format anywhere in the string, right?

**Comment by snicoll** [comment]
> Just so I understand, you're advocating removal of the emoji config option and consolidating it into a single title that the user provides, correct?

Actually I wasn't but that's a good idea. It would give you the flexibility to format titles any way you like. I am not aware of us using the emoji + title configuration option. Our usage is based on the  `ReleaseNotesSections` that define default sections statically.

>  it's more appropriate to say that the title accepts, but doesn't require, an emoji in markdown format anywhere in the string, right?

Good idea. We could generalize by saying that the title is in markdown format and that emoji are supported.

**Comment by steven-sheehy** [comment]
Thanks, I've modified the PR to remove emoji in favor of title containing all necessary information.

**Comment by LooTKinGz** [comment]
File Update

**Comment by steven-sheehy** [comment]
Any update on this?

**Comment by philwebb** [comment]
Thanks for the PR @steven-sheehy. This is now merged into master.

**Pull request closed without merging**